src/sage/manifolds/differentiable/diff_form.py

```diff
@@ -772,14 +781,19 @@ def hodge_dual(
             nondegenerate_tensor = self._vmodule._ambient_domain.metric()
 
         p = self.tensor_type()[1]
-        eps = nondegenerate_tensor.volume_form(p)
+        eps = nondegenerate_tensor.volume_form()
         if p == 0:
             common_domain = nondegenerate_tensor.domain().intersection(self.domain())
             result = self.restrict(common_domain) * eps.restrict(common_domain)
         else:
-            result = self.contract(*range(p), eps, *range(p))
+            result = self.up(nondegenerate_tensor).contract(*range(p), eps, *range(p))
```

Comment:
This should be reverted: while both versions are mathematically equivalent, the previous version is computationally more efficient. Indeed, `nondegenerate_tensor.volume_form(p)` is cached, so that computing `self.contract(*range(p), eps, *range(p))` is fast with respect to `self.up(nondegenerate_tensor).contract(*range(p), eps, *range(p))`

Reply:
Sadly, in the symplectic setting, they are not equivalent. A twist due to the skew-symmetry of the symplectic form is that simultaneously raising and lowering of a index that is summed-over gives a minus sign, i.e. `A_i^i = - A^i_i`.

If the performance is important, then I revert to using `volume_form(p)` and adding a `(-1)^p`-factor in the symplectic setting. What do you think?

Reply:
Yes performance is important, so the $(-1)^p$-factor seems the best solution.

Reply:
Done